tickets/src/main/java/no/unit/nva/publication/ticket/update/UpdateTicketHandler.java

```diff
@@ -192,13 +197,37 @@ private void updateStatus(UpdateTicketRequest ticketRequest, RequestUtils reques
     }
 
     private void publishingRequestSideEffects(PublishingRequestCase ticket,
-                                              UpdateTicketRequest ticketRequest) {
+                                              UpdateTicketRequest ticketRequest) throws ConflictException {
 
         if (COMPLETED.equals(ticketRequest.getStatus())) {
+            validateFilesForApproval(ticket);
             ticket.approveFiles().persistUpdate(ticketService);
         }
     }
 
+    private void validateFilesForApproval(PublishingRequestCase ticket) throws ConflictException {
+        if (filesForApprovalAreNotApprovable(ticket)) {
+            var fileNamesMissingMandatoryFields = getFileNamesMissingMandatoryFields(ticket);
+            throw new ConflictException(FILES_MISSING_MANDATORY_FIELDS_MESSAGE
+                                            .formatted(fileNamesMissingMandatoryFields));
+        }
+    }
+
+    private static String getFileNamesMissingMandatoryFields(PublishingRequestCase ticket) {
+        return ticket.getFilesForApproval().stream()
+                   .map(PendingFile.class::cast)
+                   .filter(PendingFile::isNotApprovable)
+                   .map(File.class::cast)
+                   .map(File::getName)
+                   .collect(Collectors.joining(","));
+    }
```

Comment:
Maybe file ids — I'm not keen on reporting data we haven't created ourselves as it presents an attack vector